src/aux-records/XpController.ts

```diff
@@ -0,0 +1,711 @@
+        | FailedResult
+    > {
+        const validationParams: Parameters<typeof this._validateUsers>[0] = {
+            xpIds: [],
+            authIds: [],
+        };
+        if ('xpId' in config.issuingUserId) {
+            validationParams.authIds.push(config.issuingUserId.xpId);
+        } else if ('userId' in config.issuingUserId) {
+            validationParams.xpIds.push(config.issuingUserId.userId);
+        } else
+            return {
+                success: false,
+                errorCode: 'invalid_request',
+                errorMessage: 'The issuing user must have an xpId or userId.',
+            };
+        if ('xpId' in config.receivingUserId) {
+            validationParams.authIds.push(config.receivingUserId.xpId);
+        } else if ('userId' in config.receivingUserId) {
+            validationParams.xpIds.push(config.receivingUserId.userId);
+        } else
+            return {
+                success: false,
+                errorCode: 'invalid_request',
+                errorMessage: 'The receiving user must have an xpId or userId.',
+            };
+
+        const batchParties = await this._validateUsers(validationParams);
+
+        if (!batchParties.success)
+            return {
+                success: false,
+                errorCode: 'user_not_found',
+                errorMessage: 'Some or all users were not found.',
+            };
+
+        let receivingUser = null;
+        let issuingUser = null;
+
+        if (
+            new Set(
+                batchParties.users.map((user) => {
+                    if (
+                        ('xpId' in config.receivingUserId &&
+                            user.id === config.receivingUserId.xpId) ||
+                        ('userId' in config.receivingUserId &&
+                            user.userId === config.receivingUserId.userId)
+                    ) {
+                        receivingUser = user;
+                    } else if (
+                        ('xpId' in config.issuingUserId &&
+                            user.id === config.issuingUserId.xpId) ||
+                        ('userId' in config.issuingUserId &&
+                            user.userId === config.issuingUserId.userId)
+                    ) {
+                        issuingUser = user;
+                    }
+                    return user.id;
+                })
+            ).size !== 2
+        )
+            return {
+                success: false,
+                errorCode: 'invalid_request',
+                errorMessage:
+                    'The issuing and receiving party cannot be the same user.',
+            };
+
+        return { success: true, receivingUser, issuingUser };
+    }
+
+    /**
+     * Get an Xp user's meta data (Xp meta associated with an auth user)
+     * Creates an Xp user for the auth user if one does not exist
+     */
+    async getXpUser(id: GetXpUserById): Promise<GetXpUserResult> {
```

Comment:
@LandonSiler This function should receive the ID of the currently logged in user in addition to the the ID of the user that the metadata is requested for. This is so we can validate who is allowed to access the metadata for a user and what amount of metadata they can access.

For example, we will eventually start including account balances with these metadata requests, but we don't want to return the account balance for user A to user B.